Doc: Improve MarkdownInput documentation (#9511)
marmelab/react-admin: docs/MarkdownInput.md

);
```

## Props

`<MarkdownInput>` accepts the following props:

| Prop                 | Required | Type      | Default    | Description                                                           |
| -------------------- | -------- | --------- | ---------- | --------------------------------------------------------------------- |

> | `source`             | Required | `string`  |            | The field name in the record                                          |

Reviewer: Props should be in alphabetic order in this table, stating with required props.
Ditto for the rest

| `fullWidth`          | Optional | `boolean` | `true`     | If `true`, the input will expand to fill the form width               |
| `label`              | Optional | `string`  |            | The label. If not set, the label is inferred from the child component |
| `helperText`         | Optional | `string`  |            | The helper text to display under the input                            |
| `previewStyle`       | Optional | `string`  | `vertical` | Markdown editor's preview style. Can be `tab` or `vertical`           |
| `height`             | Optional | `string`  | `512px`    | Markdown editor's height                                              |
| `initialEditType`    | Optional | `string`  | `wysiwyg`  | Markdown editor's initial edit type. Can be `markdown` or `wysiwyg`   |
| `useCommandShortcut` | Optional | `boolean` | `true`     | Whether use keyboard shortcuts to perform commands                    |

> `<MarkdownInput>` also accepts the [common input props](https://marmelab.com/react-admin/Inputs.html#common-input-props) and the [editor props](https://nhn.github.io/tui.editor/latest/ToastUIEditorCore) from the Toast UI editor.

Reviewer: ```suggestion
`<MarkdownInput>` also accepts the [common input props](https://marmelab.com/react-admin/Inputs.html#common-input-props) and the [editor props](https://nhn.github.io/tui.editor/latest/ToastUIEditorCore) from the [Toast UI](https://ui.toast.com/) editor.
```

## `source`

Specifies the field of the record that the input should edit. It is required.

{% raw %}

> 

Reviewer 2: No need for a newline here, and before the `{ % endraw %}`

```jsx

> <Form record={{ id: 123, title: 'Hello, world!', body: '<p>Lorem Ipsum</p>' }}>

Reviewer 2: You should use a Markdown value for the body, not an HTML value

    <MarkdownInput source="body" />
    {/* default value is "<p>Lorem Ipsum</p>" */}
</Form>
```

{% endraw %}

## `fullWidth`

You can make the markdown editor full width by setting the `fullWidth` prop to `true`:

{% raw %}

```jsx
<SimpleForm>
    <MarkdownInput source="description" fullwidth />
</SimpleForm>
```

{% endraw %}

## `helperText`

If you need to display a text below the markdown editor (usually to explain the expected data to the user), use the `helperText` prop.

{% raw %}

```jsx
<SimpleForm>
    <MarkdownInput
        source="description"
        helperText="Enter a description of the post"
    />
</SimpleForm>
```

{% endraw %}

## `previewStyle`

> You can customize the preview style by setting the `previewStyle` prop. It accepts `tab` or `vertical` and is set to `vertical` by default.

Reviewer 2: I'm not sure I understand what this prop does. You either need to explain it differently, or to add screenshots.

{% raw %}

```jsx
<SimpleForm>
    <MarkdownInput source="description" previewStyle="tab" />
</SimpleForm>
```

{% endraw %}

## `height`

The editor has a size that can be customized by setting the `height` prop. It is set to `512px` by default. You can use `px` or `%` units.

{% raw %}

```jsx
<SimpleForm>
    <MarkdownInput source="description" height="300px" />
</SimpleForm>
```

{% endraw %}

> 

Reviewer 2: Is there an easy way to make it 100% height?

## `initialEditType`

This prop allows to set the initial edit type of the editor. It accepts `markdown` or `wysiwyg` and is set to `wysiwyg` by default.

{% raw %}

```jsx
<SimpleForm>
    <MarkdownInput source="description" initialEditType="markdown" />
</SimpleForm>
```

{% endraw %}

## `useCommandShortcut`

You can disable the keyboard shortcuts by setting the `useCommandShortcut` prop to `false`. It is set to `true` by default.

{% raw %}

```jsx
<SimpleForm>
    <MarkdownInput source="description" useCommandShortcut={false} />
</SimpleForm>
```

{% endraw %}

> ## Add Your Own Buttons

Reviewer 2: ```suggestion
## Adding Buttons
```

Or
```suggestion
## `toolbarItem`
```

In any case, you need to add the `toolbarItem` prop to the Props table

> You can add your own buttons to the markdown editor by using the `toolbarItems` prop. It accepts an array of [toolbar items](https://nhn.github.io/tui.editor/latest/ToastUIEditorToolbar#toolbar-items) and is set to `null` by default.

Reviewer: Wrong link:

```suggestion
You can add your own buttons to the markdown editor by using the `toolbarItems` prop. It accepts an array of [toolbar items](https://nhn.github.io/tui.editor/latest/tutorial-example15-customizing-toolbar-buttons) and is set to `null` by default.
```

import { MarkdownInput } from "@react-admin/ra-markdown";

// Additional props are passed to `tui-editor`'s `<Editor>` component
const options = {
  previewStyle: "tab",
  height: "300px",
  initialEditType: "markdown",
  useCommandShortcut: false,
};

const PostEdit = () => (
  <Edit>
    <SimpleForm>
      <TextInput source="title" />
      <MarkdownInput source="description" {...options} />
    </SimpleForm>
  </Edit>
);
```

## Props

`<MarkdownInput>` accepts the following props:

| Prop                 | Required | Type      | Default    | Description                                                                                         |
| -------------------- | -------- | --------- | ---------- | --------------------------------------------------------------------------------------------------- |
| `source`             | Required | `string`  |            | The field name in the record                                                                        |
| `fullWidth`          | Optional | `boolean` | `true`     | If `true`, the input will expand to fill the form width                                             |

> | `helperText`         | Optional | `string`  |            | The helper text to display under the input                                                          |

Reviewer: It seems `helperText` and `height` are not in alphabetic order. Sorry

</SimpleForm>
```
{% endraw %}

## Adding Buttons

You can add your own buttons to the markdown editor by using the `toolbarItems` prop. It accepts an array of [toolbar items](https://nhn.github.io/tui.editor/latest/tutorial-example15-customizing-toolbar-buttons) and is set to `null` by default.

{% raw %}
```jsx
function createLastButton() {
    const button = document.createElement('button');
    button.className = 'toastui-editor-toolbar-icons last';
    button.style.backgroundImage = 'none';
    button.style.margin = '0';
    button.style.width = '100%';
    button.innerHTML = `<i>Custom Button</i>`;
    button.addEventListener('click', () => {
        alert('Custom Button action');
    });

    return button;
}

return (
    <SimpleForm>
        <MarkdownInput
            source="description"
            toolbarItems={[
                ['heading', 'bold', 'italic', 'strike'],
                [
                    {
                        el: createLastButton(),
                        tooltip: 'Custom Command',
                        name: 'custom',
                    },
                ],
            ]}
        />
    </SimpleForm>

> );

Reviewer 3: Can you complete this example? Imports, real component, etc.

);
```

You can customize the markdown renderer used for the preview, so that it matches the rendering you need in read mode just by applying the CSS rules you want.

> ```JSX

Reviewer 2: ```suggestion
```jsx
```

| `previewStyle`       | Optional | `string`  | `vertical` | Markdown editor's preview style. Can be `tab` or `vertical`                                         |
| `toolbarItems`       | Optional | `array`   |            | The toolbar items to display in the editor. See [Adding Buttons](#adding-buttons) for more details. |
| `useCommandShortcut` | Optional | `boolean` | `true`     | Whether use keyboard shortcuts to perform commands                                                  |

`<MarkdownInput>` also accepts the [common input props](https://marmelab.com/react-admin/Inputs.html#common-input-props) and the [editor props](https://nhn.github.io/tui.editor/latest/ToastUIEditorCore) from the [Toast UI](https://ui.toast.com/) editor.

## `source`

Specifies the field of the record that the input should edit. It is required.

{% raw %}
```jsx
<Form record={{ id: 123, title: 'Hello, world!', body: '**Lorem Ipsum**' }}>
    <MarkdownInput source="body" />
    {/* default value is "**Lorem Ipsum**" */}
</Form>
```
{% endraw %}

## `fullWidth`

You can make the markdown editor full width by setting the `fullWidth` prop to `true`:

> {% raw %}

Reviewer 2: no need for raw if no `{{` in the source

```jsx
<SimpleForm>
    <MarkdownInput source="description" fullwidth />
</SimpleForm>
```
{% endraw %}

## `height`

The editor has a size that can be customized by setting the `height` prop. It is set to `512px` by default. You can use `px` or `%` units.

> {% raw %}

Reviewer 2: no need for raw if no `{{` in the source

```jsx
<SimpleForm>
    <MarkdownInput source="description" height="300px" />
</SimpleForm>
```
{% endraw %}

## `helperText`

If you need to display a text below the markdown editor (usually to explain the expected data to the user), use the `helperText` prop.

> {% raw %}

Reviewer 2: no need for raw if no `{{` in the source

```jsx
<SimpleForm>
    <MarkdownInput
        source="description"
        helperText="Enter a description of the post"
    />
</SimpleForm>
```
{% endraw %}

## `initialEditType`

This prop allows to set the initial edit type of the editor. It accepts `markdown` or `wysiwyg` and is set to `wysiwyg` by default.

> 

Reviewer 2: no need for raw if no `{{` in the source

{% raw %}
```jsx
<SimpleForm>
    <MarkdownInput source="description" initialEditType="markdown" />
</SimpleForm>
```
{% endraw %}

## `label`

You can customize the label by setting the `label` prop. It is inferred from the `source` prop by default.

> {% raw %}

Reviewer 2: no need for raw if no `{{` in the source

```jsx
<SimpleForm>

>     <MarkdownInput source="description" label="Description" />

Reviewer 2: bad example: in this case, you don't need to set the label as it's automatically inferred.

</SimpleForm>
```
{% endraw %}

## `previewStyle`

You can customize the preview style by setting the `previewStyle` prop. It accepts `tab` or `vertical` and is set to `vertical` by default.

> If you chose `vertical`, the content and the preview will be displayed side by side. If you chose `tab`, the content and the preview will be displayed in two separate tabs. You can switch between the two tabs by clicking on it.

Reviewer 2: ```suggestion
- With the `vertical` style, the content and the preview will be displayed side by side.
- With the `tab` style, the content and the preview will be displayed in two separate tabs. Users can switch between the two tabs by clicking on the tab header.
```

> {% raw %}

Reviewer 2: no need for raw if no `{{` in the source

```jsx
<SimpleForm>
    <MarkdownInput source="description" previewStyle="tab" />
</SimpleForm>
```
{% endraw %}

## `useCommandShortcut`

You can disable the keyboard shortcuts by setting the `useCommandShortcut` prop to `false`. It is set to `true` by default.

> {% raw %}

Reviewer 2: no need for raw if no `{{` in the source

```jsx
<SimpleForm>
    <MarkdownInput source="description" useCommandShortcut={false} />
</SimpleForm>
```
{% endraw %}

## Adding Buttons

You can add your own buttons to the markdown editor by using the `toolbarItems` prop. It accepts an array of [toolbar items](https://nhn.github.io/tui.editor/latest/tutorial-example15-customizing-toolbar-buttons) and is set to `null` by default.

{% raw %}

> ```tsx

Reviewer 2: please introduced this example by explaining what you're doing.

| ----------------- | -------- | -------- | --------- | ----------------------------------------------------------------------------------------------------------------------- |
| `source`          | Required | `string`   |           | The field name in the record                                                                                            |
| `fullWidth` | Optional | `boolean` | `true` | If `true`, the input will expand to fill the form width |
| `label`      | Optional | `string`           |           | The label. If not set, the label is inferred from the child component |
| `helperText`      | Optional | `string`           |           | The helper text to display under the input |
| `previewStyle`     | Optional | `string`           |     `vertical`      | Markdown editor's preview style. Can be `tab` or `vertical` |
| `height`     | Optional | `string`           |     `512px`      | Markdown editor's height |
| `initialEditType`     | Optional | `string`           |     `wysiwyg`      | Markdown editor's initial edit type. Can be `markdown` or `wysiwyg` |

> | `useCommandShortcut`     | Optional | `boolean`           |     `true`      | Whether use keyboard shortcuts to perform commands|

Reviewer 3: Can you please reformat this?
